src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,169 @@
+            };
+            var signparames = RSA2Sign(sortedparams);
+            if (Options.NotValidateSign)
+            {
+                signparames.Remove("timestamp");
+                signparames.Remove("sign");
+            }
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, signparames);
+
+            using var response = await Backchannel.GetAsync(address);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+
+            using var alipay_user_info_share_response = JsonDocument.Parse(payload.RootElement.GetProperty("alipay_user_info_share_response").ToString());
+
+            string status = alipay_user_info_share_response.RootElement.GetString("code");
+            if (status != "10000")
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following message: {Message}.",
+                                /* Status: */ status,
+                                /* Message: */ payload.RootElement.GetString("msg"));
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, alipay_user_info_share_response.RootElement.GetString("user_id"), ClaimValueTypes.String, Options.ClaimsIssuer));
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, alipay_user_info_share_response.RootElement);
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        protected override string FormatScope() => string.Join(",", Options.Scope);
+        private IDictionary<string, string> RSA2Sign(SortedDictionary<string, string> source)
+        {
+            var textparams = source.Where(p => !string.IsNullOrEmpty(p.Value)).Select(p => p.Key + "=" + p.Value);
+            if (!textparams.Any())
+                throw new HttpRequestException("sign fail");
```

Comment:
This isn't a useful exception message for someone consuming the library to try and debug/diagnose.